compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2868,6 +2891,81 @@ public void visit(BLangRestBindingPattern restBindingPattern) {
         restBindingPattern.declaredVars.put(name.value, restBindingPattern.symbol);
     }
 
+    @Override
+    public void visit(BLangErrorBindingPattern errorBindingPattern) {
+        if (errorBindingPattern.errorTypeReference != null) {
+            errorBindingPattern.type = symResolver.resolveTypeNode(errorBindingPattern.errorTypeReference, env);
+        } else {
+            errorBindingPattern.type = symTable.errorType;
+        }
+
+        if (errorBindingPattern.errorMessageBindingPattern != null) {
```

Comment:
Can extract `errorBindingPattern.errorMessageBindingPattern` out to a variable. Applies to the cause and field patterns too.